frontend/src/components/dashboard/customHomePage/CustomHomePage.js

```diff
@@ -51,14 +98,104 @@ export class CustomHomePage extends Component {
     });
   };
 
+  handleVariancePageChange = (page, per_page) => {
+    this.setState({ variancesLoaded: false });
+    const params = { ...this.state.params, page, per_page };
+    return this.props.fetchVariances(params).then(() => {
+      this.setState({ variancesLoaded: true, params });
+    });
+  };
+
+  handleUpdateVariance = (files, variance, isApproved) => (values) => {
+    // if the application isApproved, set issue_date to today and set expiry_date 5 years from today,
+    // unless the user sets a custom expiry.
+    const { variance_document_category_code } = values;
+    const issue_date = isApproved ? moment().format("YYYY-MM-DD") : null;
+    let expiry_date;
+    if (isApproved) {
+      expiry_date = values.expiry_date
+        ? values.expiry_date
+        : moment(issue_date, "YYYY-MM-DD").add(5, "years");
+    }
+    const newValues = { ...values, issue_date, expiry_date };
+    const mineGuid = variance.mine_guid;
+    const varianceGuid = variance.variance_guid;
+    const codeLabel = this.props.complianceCodesHash[variance.compliance_article_id];
+    this.props.updateVariance({ mineGuid, varianceGuid, codeLabel }, newValues).then(async () => {
+      await Promise.all(
+        Object.entries(files).map(([document_manager_guid, document_name]) =>
+          this.props.addDocumentToVariance(
+            { mineGuid, varianceGuid },
+            {
+              document_manager_guid,
+              document_name,
+              variance_document_category_code,
+            }
+          )
+        )
+      );
+      this.props.closeModal();
+      this.props.fetchVariances(this.state.params).then(() => {
```

Comment:
I could be wrong, but this makes an API request to get the latest version of the variances, correct?

When a document is associated with a variance, the updated variance should be returned. Which means that the response from each `addDocumentToVariance` should be the latest version of the variance. That would make the following call to `fetchVariances` unnecessary.

Let me know if I'm wrong about either of these things or if you have any questions

Reply:
The call `fetchVariances` will update the `variances` redux state with the updated variance information. 
The response returned from `addDocumentToVariance` could, in theory, update the singular `variance` state, but the `fetchVariances` is necessary to update the table view of all `variances`.

Reply:
I'm not clear on why updating the singular variance state is insufficient. But I know this is the current pattern across the app, so it's probably a bit OOS anyway.

Reply:
Updating the singular 'variance' will have no effect on the same variance record that's contained in the `variances` list. 

The state for `variance` is updated every time an edit modal is opened, whereas the 'variances` state is only updated when the page is initially loaded.
 If `fetchVariances` is not called every time a record is updated than the table view will never update to reflect the change in data (unless a route change or hard refresh occurs)

Is that more clear?

Reply:
Yep